web-console/src/views/tasks-view.tsx

```diff
@@ -429,29 +458,33 @@ ORDER BY "rank" DESC, "created_time" DESC`);
             Header: "Task ID",
             accessor: "task_id",
             width: 300,
-            Aggregated: row => ''
+            Aggregated: row => '',
+            show: this.taskTableColumnSelectionHandler.showColumn("Task ID")
           },
           {
             Header: "Type",
             accessor: "type",
             Cell: row => {
               const value = row.value;
               return <a onClick={() => { this.setState({ taskFilter: addFilter(taskFilter, 'type', value) }); }}>{value}</a>;
-            }
+            },
+            show: this.taskTableColumnSelectionHandler.showColumn("Type")
           },
           {
             Header: "Datasource",
             accessor: "datasource",
             Cell: row => {
               const value = row.value;
               return <a onClick={() => { this.setState({ taskFilter: addFilter(taskFilter, 'datasource', value) }); }}>{value}</a>;
-            }
+            },
+            show: this.taskTableColumnSelectionHandler.showColumn("Datasource")
           },
           {
             Header: "Created time",
             accessor: "created_time",
             width: 120,
-            Aggregated: row => ''
+            Aggregated: row => '',
+            show: this.taskTableColumnSelectionHandler.showColumn("Created time")
```

Comment:
Just one general comment: you are doing a lot of `this.blahTableColumnSelectionHandler` in the render functions, it would be nicer to do `const { blahTableColumnSelectionHandler } = this` at the start of the function and then not do all the dereferencing. It would be slightly more performant but also will be less code :-) save all the `this.`